webapi/src/main/scala/org/knora/webapi/messages/store/triplestoremessages/TriplestoreMessages.scala

```diff
@@ -116,6 +117,21 @@ case class SparqlConstructRequest(sparql: String) extends TriplestoreRequest
   */
 case class SparqlConstructResponse(statements: Map[IRI, Seq[(IRI, String)]])
 
+/**
+  * Represents a SPARQL CONSTRUCT query to be sent to the triplestore. A successful response will be a
+  * [[SparqlExtendedConstructResponse]].
+  *
+  * @param sparql       the SPARQL string.
+  */
+case class SparqlExtendedConstructRequest(sparql: String) extends TriplestoreRequest
+
+/**
+  * A response to a [[SparqlExtendedConstructRequest]].
+  *
+  * @param statements a map of subject IRIs to statements about each subject.
+  */
+case class SparqlExtendedConstructResponse(statements: Map[IRI, Map[IRI, Seq[StringV2]]])
```

Comment:
This would then need to be more generalized. Maybe something like this?:

```
abstract case class LiteralV2
case class StringV2 extends LiteralV2
case class BooleanV2 extends LiteralV2
...

case class SparqlExtendedConstructResponse(statements: Map[IRI, Map[IRI, Seq[LiteralV2]]])
```

Reply:
Yes, I think something like that would be great.